lib/api/controllers/security.js

```diff
@@ -1480,6 +1333,176 @@ class SecurityController extends NativeController {
     });
     return securityDocument;
   }
+
+  /**
+   * @param {Request} request
+   * @param {{ _id: string, _kuzzle_info: object }} pojoUser
+   * @returns {Promise<unknown>}
+   * @private
+   */
+  async _persistUser(request, pojoUser) {
+    let loadedUser;
+
+    // Add metadata
+    pojoUser._kuzzle_info = {
+      author: request.context.user ? String(request.context.user._id) : null,
+      createdAt: Date.now(),
+      updatedAt: null,
+      updater: null
+    };
+
+    try {
+      loadedUser = await this.kuzzle.repositories.user.load(pojoUser._id);
+    }
+    catch (error) {
+      if (!(error instanceof NotFoundError)) {
+        throw error;
+      }
+      loadedUser = null;
+    }
+
+    if (loadedUser !== null) {
+      throw errorsManager.get('security', 'user', 'already_exists', pojoUser._id);
+    }
+
+    const strategies = request.input.body.credentials
+      ? Object.keys(request.input.body.credentials)
+      : [];
+
+    // Early checks before the user is created
+    for (const strategy of strategies) {
```

Comment:
Nitpicking: this could be replaced by
```suggestion
    for (const strategy of Object.keys(strategies)) {
```

Reply:
@Aschen > I don't get it, what do you mean?

Reply:
Here honestly I don't remember :sweat_smile: